chia/wallet/vc_wallet/cr_cat_wallet.py

```diff
@@ -867,28 +869,41 @@ async def match_hinted_coin(self, coin: Coin, hint: bytes32) -> bool:
         This matches coins that are either CRCATs with the hint as the inner puzzle, or CRCATs in the pending approval
         state that will come to us once claimed.
         """
-        return (
+        authorized_providers_hash: bytes32 = Program.to(self.info.authorized_providers).get_tree_hash()
+        proofs_checker_hash: bytes32 = self.info.proofs_checker.as_program().get_tree_hash()
+        hint_inner_hash: bytes32 = construct_cr_layer_hash(
+            authorized_providers_hash,
+            proofs_checker_hash,
+            hint,
+        )
+        if (
             construct_cat_puzzle(
-                CAT_MOD,
+                Program.to(CAT_MOD_HASH),
                 self.info.limitations_program_hash,
-                construct_cr_layer(
-                    self.info.authorized_providers,
-                    self.info.proofs_checker.as_program(),
-                    hint,  # type: ignore
-                ),
-            ).get_tree_hash_precalc(hint)
+                hint_inner_hash,  # type: ignore
```

Comment:
is `bytes` or `bytes32` not an acceptable type for the hint?
It looks like this would be exactly the right type here, why does the type need to be ignored here?

Reply:
We're calling the function that constructs the _puzzle_ which expects an inner _puzzle_ rather than puzzle hash. The `precalc` is making sure the hash comes out properly, but the arguments we're passing in are hashes rather than programs as it expects.